azure/datalake/store/core.py

```diff
@@ -717,16 +720,16 @@ def __exit__(self, *args):
         self.close()
 
 
-def _fetch_range(rest, path, start, end, max_attempts=10):
+def _fetch_range(rest, path, start, end, max_attempts=10, stream=False):
```

Comment:
Stream is set to true for downloads. See `get_chunk` in `multithread.py`.

Streaming is a feature of the `requests` package. Azure sends chunked responses, so I saw a large performance improvement when I switched to getting data as soon as it was available from Azure.